transfermethodui/src/main/java/com/hyperwallet/android/ui/transfermethod/view/widget/AbstractMaskedInputWidget.java

```diff
@@ -0,0 +1,143 @@
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.hyperwallet.android.ui.transfermethod.view.widget;
+
+import android.text.TextUtils;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.hyperwallet.android.model.graphql.field.HyperwalletField;
+
+/**
+ * Provides additional methods for Widget to help with formatting for input types.
+ */
+public abstract class AbstractMaskedInputWidget extends AbstractWidget {
+    private static final char NUMBER_TOKEN = '#';
+    private static final char TEXT_TOKEN = '@';
+    private static final char LETTER_OR_NUMBER_TOKEN = '*';
+
+    public AbstractMaskedInputWidget(@Nullable HyperwalletField field, @NonNull WidgetEventListener listener,
+            @Nullable String defaultValue, @NonNull View defaultFocusView) {
+        super(field, listener, defaultValue, defaultFocusView);
+    }
+
+    /**
+     * Removes the characters matching in the field's scrubRegex. This is to prepare the data for submission to the
+     * server.
+     *
+     * @param displayValue the value that would be coming from the UI component
+     * @return the raw value that will be sent to the server for submission and validation
+     */
+    protected String formatToApi(@NonNull final String displayValue) {
+        if (mField.getMask() != null) {
+            return displayValue.replaceAll(mField.getMask().getScrubRegex(), "");
+        }
+        return displayValue;
+    }
+
+    /**
+     * Uses the input and returns the formatted value specified in the mask pattern. Uses the conditional pattern as
+     * necessary.
+     *
+     * @param apiValue data in raw form
+     * @return a String formatted to the specification in the mask pattern
+     */
+    protected String formatToDisplay(@NonNull final String apiValue) {
+        if (mField != null && mField.getMask() != null) {
+            // format
+            String pattern = mField.getMask().getPattern(apiValue);
+            if (!TextUtils.isEmpty(pattern)) {
+                return format(apiValue, pattern);
+            } else {
+                return apiValue;
+            }
+        }
+        return apiValue;
+    }
+
+    String format(@NonNull final String apiValue, @NonNull final String pattern) {
+        StringBuilder formattedValue = new StringBuilder();
+        String extraTokens = "";
+        String backslash = "";
+        int patternIndex = 0;
+        int textIndex = 0;
+
+        while (true) {
+            if (textIndex >= apiValue.length() || patternIndex >= pattern.length()) {
+                break;
+            }
+            char token = pattern.charAt(patternIndex);
+            char textChar = apiValue.charAt(textIndex);
+
+            if (token == '\\') {
```

Comment:
why are we handling this use case? Is there a real case scenario where we'd have a backslash as part of the pattern? This seems pretty unrealistic to me. Let's discuss where this is coming from, cause I'm not convinced we should handle that scenario

Reply:
ok In that case, please add the "\\" as part of the constants